### PR TITLE
Add date of birth page

### PR DESCRIPTION
**Pull request by bilbof**
https://trello.com/c/VBjHzNCy/63-what-is-your-date-of-birth-question

Screenshot 2020-03-21 at 18 28 43

### CODE REVIEW

File: app/helpers/field_validation_helper.rb [hunk 1]
```diff
@@ -53,6 +53,19 @@ def validate_checkbox_field(page, values:, allowed_values:)
...
   end
 
+  def validate_date_fields(year, month, day, field)
+    return [] if year.blank? && month.blank? && day.blank?
+
+    invalid_fields = []
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_year", field: field).humanize } if year.blank?
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_month", field: field).humanize } if month.blank?
+    invalid_fields << { field: field.to_s, text: t("coronavirus_form.errors.missing_day", field: field).humanize } if day.blank?
+    unless(invalid_fields != [] || Date.valid_date?(year.to_i, month.to_i, day.to_i))
```
Comment (issyl0) [inline app/helpers/field_validation_helper.rb:63]: Do we have any steer on over 18 or under 18 validations yet?

Comment (bilbof) [reply app/helpers/field_validation_helper.rb:63]: Good question, I've asked if we want to evaluate the date beyond checking it's valid.

Comment (bilbof) [reply app/helpers/field_validation_helper.rb:63]: I'm told we shouldn't change behaviour based on the data here.

File: app/controllers/coronavirus_form/date_of_birth_controller.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    end
+  end
+
+private
+
+  PAGE = "date_of_birth"
+  NEXT_PAGE = "start" # TODO changeme
+
+  def previous_path
+    "/" # TODO
```
Comment (huwd) [inline app/controllers/coronavirus_form/date_of_birth_controller.rb]: I think contact_details has been merged now, can be this, no?
```suggestion
    coronavirus_form_contact_details_path
```

**Review by issyl0 — COMMENTED** [review]
Does it validate the date format? ie the example asks for the full year (4 numbers), does two work as well? Or are we assuming that people follow the instructions?

**Comment by bilbof** [comment]
@issyl0 I nicked that validation from the brexit funding form. Provided that we can parse the date using `Date.valid_date?` we accept it.

**Review by huwd — APPROVED** [review]

**Pull request merged**